### DISCUSSION TITLE
feat: add default model permissions for providers

### DISCUSSION THREAD

**Discussion by MRGRD56** [open-webui/open-webui#7091] [General]
After the 0.4 update all my models (that were private) became available to all users. Is there a way to fix it, besides editing the permissions of each of the models separately?

There should be a feature to make all the models from a specific provider/connection (ollama or openai one) only available to specific roles (by default)

You might want to add it in this menu

![image](https://github.com/user-attachments/assets/285be3cd-a126-4f58-9336-1dad1e21b006)

Just like you've added it here: in the model edit menu

![image](https://github.com/user-attachments/assets/f0874c84-df4b-407d-8814-8b1cacc7398b)

So you can set rights for providers, that will work for all its models by default

Otherwise, as far as I understand, now as soon as I download a new model in ollama, every user will have access to it by default, until I change its settings manually. Or when a new model comes out, in some API, it becomes public automatically, and there's no way to change it now, right? It doesn't sound really convenient

Or at least it would be nice if there was a single option to change the default permissions for new models

**Comment by fawwazanvilen** [comment]
yeah, the previous behavior was perfect for my use case. i share my open-webui instance with my friends through openrouter.ai api and i encourage them to explore the models that are available there.

now the only way to conveniently do this is by assigning them as admin - which is undesirable.

**Comment by nikallass** [comment]
I helped Sonnet 3.5 to write a script modifying Group ACLs:
https://github.com/nikallass/open-webui-acl-edit
